core/templates/domain/exploration/editable-exploration-backend-api.service.ts

```diff
@@ -16,152 +16,177 @@
-  'UrlInterpolationService', 'EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE',
-  'EDITABLE_EXPLORATION_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, ReadOnlyExplorationBackendApiService,
-      UrlInterpolationService, EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE,
-      EDITABLE_EXPLORATION_DATA_URL_TEMPLATE) {
-    var _fetchExploration = function(
-        explorationId, applyDraft, successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(
-        explorationId, applyDraft);
-      $http.get(editableExplorationDataUrl).then(function(response) {
-        var exploration = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(exploration);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableExplorationBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private readOnlyExplorationBackendApiService:
+    ReadOnlyExplorationBackendApiService,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  private _getExplorationUrl(
+      explorationId: string,
+      applyDraft: boolean): string {
+    if (applyDraft) {
+      return this.urlInterpolationService.interpolateUrl(
+        AppConstants.EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE, {
+          exploration_id: explorationId,
+          apply_draft: JSON.stringify(applyDraft)
         }
+      );
+    }
+    return this.urlInterpolationService.interpolateUrl(
+      AppConstants.EDITABLE_EXPLORATION_DATA_URL_TEMPLATE, {
+        exploration_id: explorationId
+      }
+    );
+  }
+
+  private _fetchExploration(
+      explorationId: string,
+      applyDraft: boolean,
+      successCallback:
+      (value: ExplorationBackendDict | DraftExplorationResponse) => void,
+      errorCallback: (reason?: string) => void)
+      : Promise<ExplorationBackendDict | DraftExplorationResponse> {
+    return new Promise((resolve, reject) => {
+      let editableExplorationDataUrl = this._getExplorationUrl(
+        explorationId, applyDraft);
+      this.httpClient.get<ExplorationBackendDict | DraftExplorationResponse>(
+        editableExplorationDataUrl).toPromise().then((response) => {
+        let exploration = cloneDeep(response);
+        successCallback(exploration);
+      }, (errorResponse) => {
+        errorCallback(errorResponse.error.error);
       });
-    };
+    });
+  }
 
-    var _updateExploration = function(
-        explorationId, explorationVersion, commitMessage, changeList,
-        successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(
+  private _updateExploration(
+      explorationId: string,
+      explorationVersion: string,
+      commitMessage: string,
+      changeList: string[],
+      successCallback: (value: ExplorationBackendDict) => void,
```

Comment:
Double-check the type here, please. Does it represent all the keys in the object?